jenkins-list-dependency-changes.sh

```diff
@@ -1,58 +1,58 @@
+fi
 
 # Colors for output
 CYAN='\033[0;36m'
 YELLOW='\033[1;33m'
 NC='\033[0m' # No Color
 
-# Download the older Jenkins war
-echo -e "${CYAN}Running : ${YELLOW}mvn dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion=$VERSION_ONE -Dtransitive=false${NC}"
-mvn dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion=$VERSION_ONE -Dtransitive=false
-
-# Unzip its contents to VERSION_ONE folder
-echo -e "${CYAN}Running : ${YELLOW}unzip -q ~/.m2/repository/org/jenkins-ci/main/jenkins-war/$VERSION_ONE/jenkins-war-$VERSION_ONE.war -d $VERSION_ONE${NC}\n\n"
-unzip -q ~/.m2/repository/org/jenkins-ci/main/jenkins-war/$VERSION_ONE/jenkins-war-$VERSION_ONE.war -d $VERSION_ONE
+get_war() {
+    local version="${1}"
 
-# Download the newer Jenkins war
-echo -e "${CYAN}Running : ${YELLOW}mvn dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion=$VERSION_TWO -Dtransitive=false${NC}"
-mvn dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion=$VERSION_TWO -Dtransitive=false
+    echo -e "${CYAN}Running : ${YELLOW}mvn dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion="${version}" -Dtransitive=false${NC}"
+    mvn -q dependency:get -DartifactId=jenkins-war -DgroupId=org.jenkins-ci.main -Dpackaging=war -Dversion="${version}" -Dtransitive=false
+}
 
-# Unzip its contents to VERSION_TWO folder
-echo -e "${CYAN}Running : ${YELLOW}unzip -q ~/.m2/repository/org/jenkins-ci/main/jenkins-war/$VERSION_TWO/jenkins-war-$VERSION_TWO.war -d $VERSION_TWO${NC}"
-unzip -q ~/.m2/repository/org/jenkins-ci/main/jenkins-war/$VERSION_TWO/jenkins-war-$VERSION_TWO.war -d $VERSION_TWO
+get_dependencie_list() {
+    local version="${1}"
 
-# List out the jars in /WEB-INF/lib/ for each version and find the diff
-ls -1 $VERSION_ONE/WEB-INF/lib/ > version_one_libs.txt
-ls -1 $VERSION_TWO/WEB-INF/lib/ > version_two_libs.txt
+    echo -e "${CYAN}Running : ${YELLOW}jar --list --file [jenkins-war-${version}.war]${NC}"
+    jar --list --file ${HOME}/.m2/repository/org/jenkins-ci/main/jenkins-war/${version}/jenkins-war-${version}.war | grep --extended-regexp "WEB-INF/lib/.*\.jar" | sort > ${version}_libs.txt
+}
 
-diff -u version_one_libs.txt version_two_libs.txt > lib_diff.txt
+# Download the older Jenkins war
+get_war ${VERSION_ONE}
 
-# Output the differences
-printf "\n\n\n"
+# Unzip its contents to VERSION_ONE folder
+get_dependencie_list ${VERSION_ONE}
 
-echo -e "${CYAN}$VERSION_ONE ${YELLOW}dependency:${NC}"
-cat version_one_libs.txt
+# Download the newer Jenkins war
+get_war ${VERSION_TWO}
 
-printf "\n\n"
+# Unzip its contents to VERSION_TWO folder
+get_dependencie_list ${VERSION_TWO}
 
-echo -e "${CYAN}$VERSION_TWO ${YELLOW}dependency:${NC}"
-cat version_two_libs.txt
+diff --unified=0 ${VERSION_ONE}_libs.txt ${VERSION_TWO}_libs.txt > lib_diff.txt || true
```

Comment:
weird I was running this code, for some reason the `lib_diff.txt` did not get generated 
<img width="1722" alt="image" src="https://github.com/user-attachments/assets/826edd99-d228-4f56-a956-4f1a4549bf96">

was it fine only your systems? or is it only on my system 🤔

Reply:
this is because of the last line, the `rm` removes those files.

Reply:
should be good with https://github.com/nevingeorgesunny/jenkins-core-list-dependency/pull/1/commits/f8acb2cc3f92836a87f24d7aeba526b91e04b5ed

Reply:
even better with https://github.com/nevingeorgesunny/jenkins-core-list-dependency/pull/1/commits/a0e9c8a914b54adaf804cb11ecbcbb69bda9b4cc

Reply:
let me check it out